lib/elixir/src/elixir_with.erl

```diff
@@ -68,27 +68,34 @@ translate(Meta, Args, S) ->
   CaseExpr =
     case ExprList of
       [{else, ElseExpr}] ->
-        build_else(Meta, build_cases(Parts, {ok, Expr}, fun(X) -> {error, X} end), ElseExpr);
+        build_else(Meta, build_case(Parts, {ok, Expr}, fun(X) -> {error, X} end, S), ElseExpr);
       [] ->
-        build_cases(Parts, Expr, fun(X) -> X end)
+        build_case(Parts, Expr, fun(X) -> X end, S)
     end,
   {TC, TS} = elixir_translator:translate(CaseExpr, S#elixir_scope{extra=nil}),
   {TC, elixir_scope:mergec(S, TS)}.
 
-build_cases([{'<-', Meta, [Left, Right]} | Rest], DoExpr, Wrapper) ->
-  Other = {'other', Meta, ?MODULE},
+build_case([{'<-', Meta, [Left, Right]} | Rest], DoExpr, Wrapper, S) ->
+  case Left of
+    {Name, _Meta, nil} when is_atom(Name) ->
+      Message = "this matching clause will always yield the same result",
```

Comment:
The error message should be something alongside:

    the variable "~ts" on the left side of <- inside "with" will always match. If you want to assign to a variable, use = instead of <-. The latter is reserved to match on patterns